src/propagatorconstraints/ordered.jl

```diff
@@ -0,0 +1,40 @@
+"""
+Enforces a specific order in MatchVar assignments in the grammar
+The tree is defined as a tree of `AbstractMatchNode`s. 
+Such a node can either be a `MatchNode`, which contains a rule index corresponding to the 
+rule index in the grammar of the rulenode we are trying to match.
+It can also contain a `MatchVar`, which contains a single identifier symbol.
+The order defines in what order the variable assignments should be. 
+For example, if the order is `[x, y]`, the constraint will require 
+the assignment to `x` to be less than or equal to the assignment to `y`.
+"""
+struct Ordered <: PropagatorConstraint
+    tree::AbstractMatchNode
+    order::Vector{Symbol}
+end
+
+
+"""
+Propagates the Ordered constraint.
+"""
+function propagate(c::Ordered, g::Grammar, context::GrammarContext, domain::Vector{Int})::Tuple{Vector{Int}, Vector{LocalConstraint}}
+    commutativity_constraint = LocalOrdered(context.nodeLocation, c.tree, c.order)
```

Comment:
is `commutativity_constraint` a leftover name from some previous function?

Reply:
Yes, this constraint was previously called Commutativity and I probably forgot to check the variable names when renaming